File: javascript/guessingGame.js

const prompt = require("prompt-sync")({ sigint: true });

let randNum = parseInt(Math.random() * 100);
var counter = 0;
var succeed = false;

console.log(randNum, typeof randNum);
while (counter < 10) {
    var guess = parseInt(prompt("Enter your guess: "));
    console.log(guess, typeof guess);
    if (guess > randNum) {
        console.log("Guess lower.");
    } else if (guess < randNum) {
        console.log("Guess higher.");
    }
    else {
        console.log("Congragulation; You win ${counter} attempts.");
        succeed = true;
        break;
    }
    counter++;
}

if (!succeed) {
    console.log("Sorry you didn't get it. Maybe next time!!");
}

File: golang/GuessingGame/guessingGame.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	s1 := rand.NewSource(time.Now().UnixNano())
	r1 := rand.New(s1)
	fmt.Println("I am a number between 1 to 100")
	fmt.Println("Try to guess me.")
	randNumber := r1.Intn(100)
	fmt.Printf("%v", randNumber)
	guess := 0
	counter := 0
	succeed := false

	for counter < 10 {
		fmt.Println("Enter your guess: ")
		fmt.Scanf("%d", &guess)
		if guess > randNumber {
			fmt.Println("Try lower ...")
		} else if guess < randNumber {
			fmt.Println("Try higher")
		} else {
			fmt.Println("Congrats")
			succeed = true
			break
		}
		counter++
	}

	if !succeed {
		fmt.Println("You didn't guess it. Maybe next time!!")
	}
}
